### PR TITLE
Added support to scan products with all the product identifications (#85zrjbtxf)

### LINKED ISSUE #138: Add support to scan products with all the product identifications
Currently, inventory count app has the support to scan product with SKU only. We will need the support to scan the product with Barcode(ISBN,UPC,GTIN) and Shopify product ID.
Screen Shot 2023-01-17 at 12 22 28 PM

Due date: 28th Feb 2023

### PR DESCRIPTION
**Pull request by k2maan**
### Related Issues

Closes #138 

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/DivanteLtd/vsf-capybara/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: src/store/modules/product/actions.ts [hunk 1]
```diff
@@ -19,12 +19,22 @@ const actions: ActionTree<ProductState, RootState> = {
...
+        } 
+      } else {
+        params = {
+          "viewSize": payload.viewSize,
+          "viewIndex": payload.viewIndex,
+          "filters": ['isVirtual: false'],
+          "keyword":  '*' + payload.queryString + '*'
+        }
+      }
+      resp = await ProductService.fetchProducts(params);
```
Comment (disha1202) [inline src/store/modules/product/actions.ts:37]:
```suggestion
 resp = await ProductService.fetchProducts({
        "filters": [`sku: ${payload.scannedValue} OR upc: ${payload.scannedValue}`,'isVirtual: false'],
        "viewSize": payload.viewSize,
        "viewIndex": payload.viewIndex
      })
```

File: src/views/Search.vue [hunk 2]
```diff
@@ -100,7 +101,7 @@ export default defineComponent({
         .then((result) => {
           //result : value of the scanned barcode/QRcode
           if(result.role){
-            this.queryString = result.role
+            this.scannedValue = result.role;
```
Comment (disha1202) [inline src/views/Search.vue:104]:
```suggestion
            this.queryString = result.role;
```

File: src/views/Search.vue [hunk 3]
```diff
@@ -119,15 +120,19 @@ export default defineComponent({
...
+    },
+    async getProducts(vSize?: any, vIndex?: any) {
       const viewSize = vSize ? vSize : process.env.VUE_APP_VIEW_SIZE;
       const viewIndex = vIndex ? vIndex : 0;
       const payload = {
         viewSize,
         viewIndex,
-        queryString: '*' + this.queryString + '*'
+        queryString: this.queryString,
+        scannedValue: this.scannedValue,
```
Comment (disha1202) [inline src/views/Search.vue:133]:
```suggestion
        queryString: this.queryString,
```

File: src/views/Search.vue [hunk 4]
```diff
@@ -119,15 +120,19 @@ export default defineComponent({
...
       }
-      if (this.queryString) {
+      if (this.queryString || this.scannedValue) {
```
Comment (disha1202) [inline src/views/Search.vue:135]:
```suggestion
      if (this.queryString) {
```

File: src/views/Search.vue [hunk 5]
```diff
@@ -119,15 +120,19 @@ export default defineComponent({
         event.target.complete();
       })
     },
-    async getProducts(vSize: any, vIndex: any) {
+    searchProducts() {
+      this.getProducts(process.env.VUE_APP_VIEW_SIZE, 0);
+    },
```
Comment (disha1202) [inline src/views/Search.vue:125]:
```suggestion
    searchProducts() {
      this.queryString = '*' + this.queryString + '*';
      this.getProducts(process.env.VUE_APP_VIEW_SIZE, 0);
    },
```

**Review by disha1202 — CHANGES_REQUESTED** [review]

**Comment by ymaheshwari1** [comment]
Another approach is to define one more variable `searchString` that will be used to display the text in searchbar `v-model="searchString"` and when we are scanning a product assign both `queryString` and `searchString` with `result.role` and when searching using `searchbar` assign `queryString = *<valuue>* and assign `searchString = <value>`.

In both the case we can simply call `getProducts` method and we don't need to define another method.
Also here we are using two variables as if we will update the same variable used in `v-model` then `*  *` will be appended in the searchbar text as well.

File: src/views/Search.vue [hunk 6]
```diff
@@ -6,7 +6,7 @@
       </ion-toolbar>
     </ion-header>
     <ion-content>
-      <ion-searchbar @ionFocus="selectSearchBarText($event)" v-model="queryString" :placeholder="$t('Search')" v-on:keyup.enter="queryString = $event.target.value; getProducts()"/>
+      <ion-searchbar @ionFocus="selectSearchBarText($event)" v-model="queryString" :placeholder="$t('Search')" v-on:keyup.enter="queryString = $event.target.value; searchProducts()"/>
```
Comment (ymaheshwari1) [inline src/views/Search.vue:9]:
```suggestion
      <ion-searchbar @ionFocus="selectSearchBarText($event)" v-model="queryString" :placeholder="$t('Search')" @keyup.enter="queryString = $event.target.value; searchProducts()"/>
```

File: src/views/Search.vue [hunk 7]
```diff
@@ -81,7 +81,7 @@ export default defineComponent({
   },
   data (){
     return {
-      queryString: ''
+      queryString: '',
```
Comment (ymaheshwari1) [inline src/views/Search.vue:84]: Remove unwanted changes

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

**Review by ymaheshwari1 — APPROVED** [review]

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**